rules/analysis_tests/identical_outputs_test.bzl

```diff
@@ -0,0 +1,82 @@
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+
+_TestFiles = provider(
+    fields = {
+        "files": "A glob of files collected for later assertions",
+    },
+)
+
+def _identical_outputs_test_impl(ctx):
+    env = analysistest.begin(ctx)
+    all_files = []
+    asserts.true(env, len(ctx.attr.deps) > 1)
```

Comment:
The test doesn't assert the problem to have 1 dev because it asserts identical outputs amongst multiple `deps` https://github.com/bazel-ios/rules_ios/pull/333/files/5138482e5b1fcd470f76b793ef40e6ebc930cc73#diff-fa2c728306ed72d291b632ffa2b0458ea872ba83445c911667ac90ff0c4263a0R56

This is the only way to ensure that https://github.com/bazelbuild/bazel/issues/12171 isn't happening internal to a rule!